lib/gslab_model/ModelData.R

```diff
@@ -0,0 +1,190 @@
+ModelData <- setRefClass(Class  = "ModelData",
+                         # A ModelData object stores dataset in the field var, an R data.frame object.
+                         # The constructor ModelData() accepts the same inputs as the constructor for
+                         #   the R data.frame class, plus an option of specifying variable names and
+                         #   an option of adding a customer list of constants characterizing the data.
+                         
+                         fields = list(var                = "data.frame",   # a data.frame object to hold dataset
+                                       varnames           = "character",    # a vector of variable names in the dataset
+                                       nvars              = "numeric",      # the number of variables in the dataset
+                                       nobs               = "numeric",      # the number of observations in the dataset
+                                       const              = "list",         # a list to hold constants characterizing the dataset
+                                       groupvar           = "numeric",      # a vector identifying groups in the panel
+                                       ngroup             = "numeric",      # the number of groups in the panel
+                                       group_size         = "numeric",      # a vector of size ngroups giving the number of observations in each group
+                                       unique_group_sizes = "numeric"       # unique group sizes that appear in the dataset
+                         ),
+                         methods = list(
+                             initialize = function(..., varnames = NULL, const = list()) {
```

Comment:
Yes... And in the package setup, the methods defined in the `setRefClass` call won't appear in the method list.